0.10 release (#179)
flowforge/website: src/blog/2022/09/flowforge-010-released.md

---
title: FlowForge 0.10 released
subtitle: Secure HTTP end points, Read-only users and Static outbound IPs
description: Secure Node-RED HTTP end points, Read-only users and Static outbound IPs
date: 2022-09-30 12:00:00.0
authors: ["rob-marcer"]
video: d23Pmyc0k7I
---

Secure your HTTP endpoints, create read-only users in your teams and use our static IP address for outbound traffic

<!--more-->

Keep reading for the details of what's in this release or you can watch our 1 minute roundup video of the new release above. 

We're pleased to announce version 0.10 is now available. The next release of the FlowForge application containing new features, a number of improvements, and bug fixes. Keep reading for a promotion code to get your first month free on FlowForge Cloud. 

## Features
[Secure HTTP Endpoints](https://github.com/flowforge/flowforge/issues/578)
We've added the ability for you to secure your HTTP endpoints. You can now control who can access Dashboards or API endpoints you create in FlowForge.

[Read-only Users](https://github.com/flowforge/flowforge/issues/657)

> We've added a new user role for Read-only access. This will allow users to login to your FlowForge project and view the Node-RED flows including the debug output without them being able to edit anything.

Reviewer: Might be worth rewording this as they can view anything in the project not just the debug output, its just that they can't make changed

[Static Outbound IP Addresses](https://github.com/flowforge/CloudProject/issues/59)

> We've updated FlowForge Cloud so that all outbound traffic from your projects now comes from a single IP address. When trying to access a remote resource such as a database it is often a requirement for the IP address the traffic comes from to be known remain unchanged. 

Reviewer: ```suggestion
We've updated FlowForge Cloud so that all outbound traffic from your projects now comes from a single IP address. When trying to access a remote resource such as a database it is often a requirement for the IP address the traffic comes from to be fixed. 
```

## Improvements
We've made a number of improvements to the overall experience of running FlowForge.

- Allow both key and component in a ff-data-table column definition [#43](https://github.com/flowforge/forge-ui-components/issues/43)
- Default Stack and Templates [#989](https://github.com/flowforge/flowforge/issues/989)
- Provide platform containers and base stack container for administrators [#917](https://github.com/flowforge/flowforge/issues/917)

## Bug Fixes
We've fixed the following bugs in this release.

> - [Provide platform containers and base stack container for administrators](https://github.com/flowforge/flowforge/issues/917)

Reviewer: Move to improvements

- [User names can be same (but different case)](https://github.com/flowforge/flowforge/issues/983)

> - [/api/v1/user/projects endpoint broken](https://github.com/flowforge/flowforge/issues/982)

Reviewer: ```suggestion
```
This was just internal cleanup

- [User list not refreshing after changing user details](https://github.com/flowforge/flowforge/issues/463)
- [Navigating directly to a device page gets the wrong team selected](https://github.com/flowforge/flowforge/issues/986)
- [Node-RED Isn't ready when FlowForge app says it is running following a project restart](https://github.com/flowforge/flowforge/issues/941)
- [Invitations left for deleted teams](https://github.com/flowforge/flowforge/issues/923)
- [Following email verification link twice throws error](https://github.com/flowforge/flowforge/issues/1024)

> - [package does not have issues or homepage URLs](https://github.com/flowforge/flowforge-device-agent/issues/32)

Reviewer: ```suggestion
```
Just internal housekeeping

> - [Agent does not log stderr from the Node-RED process](https://github.com/flowforge/flowforge-device-agent/issues/21)

Reviewer: Move to improvements
